src/pages/pupils/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug]/index.tsx

```diff
@@ -177,18 +184,35 @@ export const getStaticProps: GetStaticProps<
         };
       }
 
-      let transcriptSentences = curriculumData.transcriptSentences;
-
-      if (curriculumData.videoTitle && !curriculumData.isLegacyLicense) {
-        // For new content we need to fetch the captions file from gCloud and parse the result to generate
-        // the transcript sentences.
-        const fileName = `${curriculumData.videoTitle}.vtt`;
-        transcriptSentences = (await getCaptionsFromFile(fileName)) ?? [];
-      }
+      // For new content we need to fetch the captions file from gCloud and parse the result to generate
+      // the transcript sentences.
+      const resolveTranscriptSentences = (() => {
+        if (curriculumData.videoTitle && !curriculumData.isLegacy) {
+          return getCaptionsFromFile(`${curriculumData.videoTitle}.vtt`);
+        }
+
+        return curriculumData.transcriptSentences;
+      })();
```

Comment:
I originally had this as a big ternary but it was unreadable.

Reply:
nice. I haven't seen self executing functions in ages but it's nice way of keeping variables as const without ternaries.

Reply:
It's an affordance I miss from my days as a Rubyist